master/buildbot/worker/protocols/manager/msgpack.py

```diff
@@ -129,8 +127,7 @@ def initialize(self) -> InlineCallbacksType[None]:
         finally:
             eventually(dispatcher.master.initLock.release)
 
-    @defer.inlineCallbacks
-    def call_update(self, msg: dict[str, Any]) -> InlineCallbacksType[None]:
+    def call_update(self, msg: dict[str, Any]) -> Deferred[None]:
```

Comment:
This should _really_ stay inlineCallbacks just for consistency, because with Deferreds it's easy to lose thrown exceptions. Now if usage is not consistent, then it's very easy not to return a Deferred where one should have been returned. Best to just use inlineCallbacks always.

Reply:
My point was that `remote_update_msgpack` does not return Deferred _yet_, but could conceivably do so in the future like other `remote_` methods. I think it makes sense to code defensively until we migrated to async/await and got warnings whenever an awaitable was not awaited. 

I think I remember noticing the exact same problem in one of this class methods during development, this is why I'm cautious here.

Reply:
Agreed.